compare50/__main__.py

```diff
@@ -100,6 +91,22 @@ def get_all(self, paths, preprocessor, is_archive=False):
                 _api.get_progress_bar().update()
         return subs
 
+    @staticmethod
+    def _is_valid_utf8(file_path):
+        """
+        Check if file_path is valid utf-8.
+        f.read() is not performant since the entire file is read in before checking if it is valid utf8.
+        This function reads in the file in increasingly large blocks so that it can error out early if necessary.
+        """
+        try:
+            with open(file_path) as f:
+                blocksize = 64
+                while f.read(blocksize):
+                    blocksize *= 2;
+        except UnicodeDecodeError:
+            return False
+        return True
```

Comment:
Would that mistakenly fail if at the end of a block there's only part of a Unicode character?

Reply:
The argument passed to read seems to correspond to codepoints rather than bytes if the file is opened in unicode mode, so not I don't think so. For example,
```
with open("test.txt", "w") as f: 
    f.write("😀")

with open("test.txt", "r") as f:
    print(f.read(1)) # prints 😀

with open("test.txt", "rb") as f:
    f.read(1).decode() # throws UnicodeDecodeError

```

Reply:
Ah, interesting! Should we pass `encoding='UTF-8'` to `open` then? It appears that that's not necessarily the default.

Reply:
oh yeah, i forgot that is locale dependent. yep, for sure